File: 01_hash/q2_전화번호 목록.py
# 20210518
def solution(phone_book):
    d = dict()
    for n in phone_book:
        d[n] = 0
    for n in phone_book:
        t = ""
        for c in n[:-1]:
            t += c
            if (t not in d):
                return False
    return True

File: 01_hash/q2_전화번호목록.cpp
def solution(phone_book):
    answer = true
    phone_book.sort()
    print(phone_book)
    for i in range(len(phone_book)-1):
        if phone_book[i] == phone_book[i+1][:len(phone_book[i])]:
            answer = false
            break
    return answer

phone_book = ["12","123","1235","567","88"]
print(solution(phone_book))